onnxruntime/core/optimizer/matmul_transpose_fusion.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+#include "core/optimizer/matmul_transpose_fusion.h"
+#include "core/graph/graph_utils.h"
+#include <deque>
+
+using namespace ONNX_NAMESPACE;
+using namespace ::onnxruntime::common;
+namespace onnxruntime {
+
+static std::pair<bool, Node*> IsInputTranspose(const Graph& graph, NodeArg& node_arg) {
+  auto trans_node = graph.GetProducerNode(node_arg.Name());
+  if (trans_node == nullptr || trans_node->OpType() != "Transpose") {
+    return std::make_pair(false, nullptr);
+  }
+
+  auto perms = RetrieveValues<int64_t>(trans_node->GetAttributes().at("perm"));
+  int64_t rank = perms.size();
+  if (rank < 2) {
+    return std::make_pair(false, nullptr);
+  }
+
+  bool is_trans_on_last_two_dims = true;
+  for (int64_t i = 0; i < rank - 2; i++) {
+    if ((int64_t)perms[i] != i) {
+      is_trans_on_last_two_dims = false;
```

Comment:
nit: coding standards say no c-style casts. use static_cast, although not sure why a cast is necessary here as perms is int64_t isn't it?

Reply:
still a couple of c-style casts of perms on line 34

---
In reply to: [397603952](https://github.com/microsoft/onnxruntime/pull/3174#discussion_r397603952) [](ancestors = 397603952)